File: backend/src/main/java/com/librarymanagement/service/DashboardService.java
package com.librarymanagement.service;

import com.librarymanagement.dto.DashboardDTO;
import com.librarymanagement.dto.UserDashboardDTO;

public interface DashboardService {
    DashboardDTO getDashboardStats();
    UserDashboardDTO getUserDashboardStats(Long userId);
}

File: frontend/src/services/dashboardService.js
import api from '../api';

const dashboardService = {
  getDashboardStats: async () => {
    const response = await api.get('/dashboard');
    return response.data;
  },

  getUserDashboardStats: async (userId) => {
    const response = await api.get(`/dashboard/user/${userId}`);
    return response.data;
  }
};

export default dashboardService;
